### PR TITLE
Support for building DVS as Docker Managed Plugin.

### PR DESCRIPTION
**Pull request by msterin**
See issue #886 
Docker managed plugins are currently documented [here](https://docs.docker.com/engine/extend/)

This PR introduces new folder `./plugin`. Running `make all` in this folder will assume that
the DVS is pre-built and dockerhub is authenticared with `docker login`, then it will  package DVS
as a Managed Plugin, and push to dockerhub.

**Note that this is not integrated with Build or CI - since we do Build under Docker, a bit more work is needed in central Makefile and drone script - I will do it in a separate PR if this one will be approved/merged soon, or in an extension to this one if it will still be in review.** 

See Makefile for name/tag and location.

Assuming the Makefile defines names as 'cnastorage/docker-volume-vsphere:0.12', the plugin  could
be installed on Docker (1.13+) as follows:

* no question asked, and pretend the plugin name is 'vsphere' (can be used in `volume create` and `plugin rm`)
```
docker plugin install --grant-all-permissions --alias vsphere cnastorage/docker-volume-vsphere:0.12
```
* vanilla interactive install as disabled plugin (will require `docker plugin enable `  to be operational
```
docker plugin install --disable cnastorage/docker-volume-vsphere:0.12
```

The `docker volume` command behaves as before but the plugin name in `docker create` is either what was passed as `--alias` or the full plugin name , i.e. `cnastorage/docker-volume-vsphere:0.12`

### CODE REVIEW

File: plugin/Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+USER := cnastorage
+
+PLUGIN_NAME=$(USER)/docker-volume-vsphere
+PLUGIN_TAG=0.12
+
+# Binaries we want to pick up
+BINARY := docker-volume-vsphere
+BIN_LOC := ../build
+
+# Tmp docker image used to constuct rootfs + our banaries
```
Comment (pdhamdhere) [inline plugin/Makefile]: typo banaries => binaries

File: plugin/config.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+	"Description": "VMWare vSphere Docker Volume plugin",
+	"Documentation": "http://vmware.github.io/docker-volume-vsphere/documentation",
+	"Entrypoint": ["/usr/bin/docker-volume-vsphere", "--log_level", "info"],
+	"PropagatedMount": "/mnt/vmdk",
+	"Mounts": [
```
Comment (pdhamdhere) [inline plugin/config.json]: What about "/etc/" for `/etc/docker-volume-vsphere.conf`

File: plugin/Makefile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# Makefile for Docker data volume *managed* plugin construction
+# This Makefile assumes it's in ./plugin in the tree and the binaries are already built
+# Eventually we will throw away deb/rpm build from ../vmdk_plugin/Makefile and will use only
+# managed plugin
+
+# User component of plugin name (and dockerhub group to push to)
+USER := cnastorage
+
+PLUGIN_NAME=$(USER)/docker-volume-vsphere
+PLUGIN_TAG=0.12
```
Comment (pdhamdhere) [inline plugin/Makefile]: Later we will have to find a way to separate dev builds from release tags.

File: plugin/config.json (hunk 2 above)
Comment (msterin) [inline plugin/config.json]: yeah. I did not want to open up all important folders. I think I will change it to /etc/vmware/<config> and open /etc/vmware. Comments ?

**Pull request merged**